radix-engine/src/engine/track.rs

```diff
@@ -233,10 +247,13 @@ impl<'s, R: FeeReserve> Track<'s, R> {
 
         match &loaded_substate.lock_state {
             LockState::Read(n) => loaded_substate.lock_state = LockState::Read(n - 1),
-            LockState::Write => loaded_substate.lock_state = LockState::no_lock(),
+            LockState::Write => {
+                loaded_substate.lock_state = LockState::no_lock();
+                loaded_substate.metastate = SubstateMetaState::Updated;
+            }
         }
 
-        if write_through {
+        if unmodified_base {
```

Comment:
I think `unmodified_base` only capture one aspect - the substate must pre-exist and unchanged. It does not indicate that any change must be immediately committed.

How about two flags `unmodified | write_through`?

Reply:
hrm good point

Reply:
what if we called it `force_write` instead

Reply:
Sounds good.